### PR TITLE
rpc: Disallow sendtoaddress and sendmany when private keys disabled

### LINKED ISSUE #21104: Watchonly descriptor wallets classifying funds as `mine`, are "spendable"
**Expected behavior**

I'd expect funds in a watchonly wallet to be shown as "watchonly", be unspendable.

**Actual behavior**

Funds are shown as "mine" in getbalances, can be "spent" (sendtoaddress completes without error, however transaction is unspent and fails to enter mempool. Balance is incorrectly decremented)

**To reproduce**

Make a watchonly descriptor wallet.

```
bitcoin-cli createwallet w2 true true "" true true false 
```

Load up a descriptor that's got some funds on it.

```
$ bitcoin-cli -rpcwallet=w2 importdescriptors '[{"desc":"wpkh(tpubD6NzVbkrYhZ4Wh1z2cvkzQsM12k9kvyMMrGXi7rdk5eyUYUFND1qhB1zEZ3pJwSSSGLYsKfZLKNvs4L7oqNb8vJj3zJTvxMRCKWsNhX9JXQ/*)#86kup682", "timestamp":"now","active":true,"range":100}]'

```

See the balance.

```
$ bitcoin-cli -rpcwallet=w2 getbalances
{
  "mine": {
    "trusted": 3.00000000,
    "untrusted_pending": 3.00000000,
    "immature": 0.00000000,
    "used": 0.00000000
  }
}
```

Spend the balance.

```
$ bitcoin-cli -rpcwallet=w2 sendtoaddress bcrt1qnk09yy5ayyhwvfydg2x58zwy443pjx5uesg9gw 3.0 "" "" true
90bce182f476376e6cf3552e234c686ce5bfee4d47739081f5cba90de374156b
```

See the balance update.

```
$ bitcoin-cli -rpcwallet=w2 getbalances
{
  "mine": {
    "trusted": 0.00000000,
    "untrusted_pending": 0.00000000,
    "immature": 0.00000000,
    "used": 3.00000000
  }
}
```

Not in the mempool though

```
$ bitcoin-cli getmempoolentry 90bce182f476376e6cf3552e234c686ce5bfee4d47739081f5cba90de374156b
error code: -5
error message:
Transaction not in mempool
```

**System information**

Bitcoin Core version v0.21.0

### PR DESCRIPTION
**Pull request by achow101**
Since `sendtoaddress` and `sendmany` (which use the `SendMoney` function) create and commit a transaction, they should not do anything when the wallet does not have private keys. Otherwise a valid transaction cannot be made.

Fixes #21104

### CODE REVIEW

**Review by jonatack — COMMENTED** [review]
ACK modulo test coverage. Here's a regression test commit that fails on master and passes with this patch https://github.com/jonatack/bitcoin/commit/15a9bcc75d5447d85b614e2e5595d7a9a8b423a6

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -409,7 +415,7 @@ UniValue SendMoney(CWallet* const pwallet, const CCoinControl &coin_control, std
     bilingual_str error;
     CTransactionRef tx;
     FeeCalculation fee_calc_out;
-    bool fCreated = pwallet->CreateTransaction(recipients, tx, nFeeRequired, nChangePosRet, error, coin_control, fee_calc_out, !pwallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS));
+    bool fCreated = pwallet->CreateTransaction(recipients, tx, nFeeRequired, nChangePosRet, error, coin_control, fee_calc_out, true);
```
Comment (jonatack) [inline src/wallet/rpcwallet.cpp:418]: while touching this line
```suggestion
    const bool fCreated = pwallet->CreateTransaction(recipients, tx, nFeeRequired, nChangePosRet, error, coin_control, fee_calc_out, true);
```

Comment (achow101) [reply src/wallet/rpcwallet.cpp:418]: Done

**Comment by achow101** [comment]
Included @jonatack's test

**Review by jonatack — COMMENTED** [review]
ACK 6bfbc97d716faad38c87603ac6049d222236d623

**Review by meshcollider — COMMENTED** [review]
utACK 6bfbc97d716faad38c87603ac6049d222236d623

**Review by kristapsk — APPROVED** [review]
ACK 6bfbc97d716faad38c87603ac6049d222236d623. "Error: Private keys are disabled for this wallet" is definitely a better error message than "Insufficient funds" here. Hopefully change of error code from -6 to -4 doesn't break any software using Bitcoin JSON-RPC API.

**Review by Escobar181 — APPROVED** [review]

**Comment by MarcoFalke** [comment]
You forgot to fix `send`?

Currently prints `error message: 'Transaction needs a change address, but we can't generate it. Please call keypoolrefill first.'.`

**Comment by achow101** [comment]
> You forgot to fix `send`?
> 
> Currently prints `error message: 'Transaction needs a change address, but we can't generate it. Please call keypoolrefill first.'.`

No, that's intended and expected if there are no change addresses. `send` is supposed to work when private keys are disabled; in that case, it will create and output a psbt.

**Comment by MarcoFalke** [comment]
Ah, ok thx.

Also, tagged for `0.21.1` backport. Let me know if this shouldn't be backported.

**Comment by MarcoFalke** [comment]
Backported in #20901

**Pull request merged**